### PR TITLE
Fix xSmbShare resource does not take into account properties other than Name, Path, Ensure when doing a Test-TargetResource

### PR DESCRIPTION
**Pull request by SuperMarioo**

### CODE REVIEW

**Comment by vors** [comment]
We would need to ask original resource authors review this change, it's quite big.

File: DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1 [hunk 1]
```diff
@@ -1,3 +1,641 @@
...
+            }
+            elseif ($access.AccessRight -eq 'Read' -and $access.AccessControlType -eq 'Allow')
+            {
+                $CurrentreadAccess += $access.AccountName
+            }            
+            elseif ($access.AccessRight -eq 'Full' -and $access.AccessControlType -eq 'Allow')
+            {
+                $CurrentfullAccess += $access.AccountName
+            }
+            elseif ($access.AccessRight -eq 'Full' -and $access.AccessControlType -eq 'Deny')
```
Comment (TravisEz13) [inline DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1:78]: deny can be at other level than  full

File: DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1 [hunk 2]
```diff
@@ -1,3 +1,641 @@
...
+$PSBoundParameter = $PSBound.Keys.Split('"')
+
+
+$OtherParamResult  = @()
+
+
+for ($i = 0; $i -lt $PSBoundParameter.Count ; $i++)
+{ 
+ 
+ $test1 = ($PSBound[$PSBoundParameter[$i]]-join ' ').ToString()
```
Comment (TravisEz13) [inline DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1:618]: give test1 and test2 meaningful names like currentvalue and expectedvalue

File: DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1 [hunk 3]
```diff
@@ -1,3 +1,641 @@
...
+
+
+
+}#end IF
+
+If   ($PSBound.ContainsKey('ChangeAccess')){
+
+     for ($i = 0; $i -lt $numberChangeAccess; $i++)
+    { 
+     $ErrorActionPreference = "SilentlyContinue"
```
Comment (TravisEz13) [inline DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1:409]: Why is silently continue needed?

File: DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1 [hunk 4]
```diff
@@ -1,3 +1,641 @@
...
+    
+    } #end IF
+ 
+$AddingUserPermissiontoObject = @{}
+
+## Creating Object for with user and permission that needs to be assigned 
+
+$usersadd = ($AddingUserPermission.GetEnumerator() | Where-Object { $_.value -eq "False"}).name 
+
+foreach ( $user in $usersadd){
```
Comment (TravisEz13) [inline DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1:440]: Feedback: It's difficult to read this code without proper indentation and with all the extra newlines.
Updated the contribution guidelines to include some of the style guidance for these issues:
[Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)

Comment (SuperMarioo) [inline DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1:440]: Thanks for feedback TravisEz13 
 I'll do  some corrections .

**Comment by TravisEz13** [comment]
Please re-open/open a new pull request when you have an update.

**Pull request closed without merging**